modules/tinymce/src/plugins/table/main/ts/api/Commands.ts

```diff
@@ -49,6 +51,93 @@ const registerCommands = (editor: Editor, actions: TableActions, cellSelection:
     });
   });
 
+  const toggleColumnHeader = () => {
+    getSelectionStartCell(editor).each((startCell) => {
+      TableLookup.table(startCell, isRoot).filter(Fun.not(isRoot)).each((table) => {
+        if (isEntireColumnsHeaders(editor, selections)) {
+          editColumnsHeader(table, startCell, 'td');
+        } else {
+          editColumnsHeader(table, startCell, 'th');
+        }
+
+        Events.fireTableModified(editor, table.dom, Events.structureModified);
+      });
+    });
+  };
+
+  const editColumnsHeader = (table: SugarElement<HTMLTableElement>, startCell: SugarElement<HTMLTableCellElement>, changeTo: 'th' | 'td'): void => {
+    const warehouse = Warehouse.fromTable(table);
+    const targets = TableTargets.forMenu(selections, table, startCell);
+
+    const usedColumns: number[] = [];
+    Arr.each(warehouse.all, (row) => {
+      Arr.each(row.cells, (cell) => {
+        const existsInSelection = Arr.exists(targets.selection, (element) => {
+          return element.dom === cell.element.dom;
+        });
+
+        if (existsInSelection && !Arr.contains(usedColumns, cell.column)) {
+          usedColumns.push(cell.column);
+        }
+      });
+    });
+
+    Arr.each(usedColumns, (value) => {
+      Arr.each(warehouse.all, (row) => {
+        const columnCell = Arr.find(row.cells, (cell) => {
+          return cell.column === value;
+        });
+
+        columnCell.each((cell) => {
+          const theadOpt = SelectorFind.ancestor(cell.element, 'thead');
+
+          if (theadOpt.isNone() && Util.getNodeName(cell.element.dom) !== changeTo) {
+            const newCellElm = editor.dom.rename(cell.element.dom, changeTo) as HTMLTableCellElement;
+            Events.fireNewCell(editor, newCellElm);
+          }
+        });
+      });
+    });
+  };
+
+  const toggleRowHeader = () => {
+    getSelectionStartCell(editor).each((startCell) => {
+      TableLookup.table(startCell, isRoot).filter(Fun.not(isRoot)).each((table) => {
+        const getNewType = () => {
+          if (isEntireRowsHeaders(editor, selections)) {
+            return 'body';
+          } else {
+            return 'header';
+          }
+        };
+
+        actions.setTableRowType(editor, {
+          type: getNewType(),
+        });
+
+        Events.fireTableModified(editor, table.dom, Events.structureModified);
+      });
+    });
+  };
+
+  const toggleCaption = (_ui: boolean, toggleState: boolean, forced?: boolean) => {
```

Comment:
Adding a `state`/`forced` property goes against all current toggle based API design in TinyMCE. Can we reconsider how this is being done? Maybe it should be a separate command entirely to clarify that, eg `mceTableAddCaption`?